File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size(), i;
        map<int, int> mp;
        vector<int>v;

        for(i=0;i<n;i++){
            if(mp.find(target - nums[i]) == mp.end()){
                mp[nums[i]] = i;
            }else{
                v.push_back(mp[target - nums[i]]);
                v.push_back(i);
                break;
            }
        }

        return v;



    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> mp = new HashMap<Integer,Integer>();
        int i,j;
        int[] result = new int[2];
        for(i=0;i<nums.length;i++) {
            mp.put(nums[i],i);
        }
        int diff;
        for(i=0;i<nums.length;i++) {
            diff=target-nums[i];
            if(mp.get(diff) != null && mp.get(diff) != i) {
                result[0]=i;
                result[1]=mp.get(diff);
                return result;
            }
        }
        return result;
    }
}